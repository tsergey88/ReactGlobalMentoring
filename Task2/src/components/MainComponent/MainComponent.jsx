import React, { Component } from "react";

export class MainComponent extends Component {
  render() {
    return (
      <div>
        <h1>Hello World</h1>
      </div>
    );
  }
}
