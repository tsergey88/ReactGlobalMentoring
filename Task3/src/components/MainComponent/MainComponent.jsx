import React from "react";
import { Header } from "../Header";
import { Footer } from "../Footer";
import {FilmsList} from "../FilmsList";

export class MainComponent extends React.Component {
  render() {
    return (
      <>
        <Header />
        <FilmsList />
        <Footer />
      </>
    );
  }
}
