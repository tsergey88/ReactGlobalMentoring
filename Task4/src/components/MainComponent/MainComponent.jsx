import React from "react";
import { Header } from "../Header";
import { Footer } from "../Footer";
import { FilmsList } from "../FilmsList";
import { ErrorBoundary } from "../ErrorBoundary";
import { SearchForm } from "../SearchForm";
import { StateLine } from "../StateLine";

export class MainComponent extends React.Component {
  render() {
    return (
      <ErrorBoundary>
        <Header>
          <SearchForm />
        </Header>
        <div>
          <StateLine />
          <FilmsList />
        </div>
        <Footer />
      </ErrorBoundary>
    );
  }
}
