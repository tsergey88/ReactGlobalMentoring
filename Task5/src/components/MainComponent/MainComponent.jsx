import React from "react";

import { Header } from "../Header";
import { Footer } from "../Footer";
import { FilmsList } from "../FilmsList";
import { ErrorBoundary } from "../ErrorBoundary";
import { SearchForm } from "../SearchForm";

export const MainComponent = () => (
  <ErrorBoundary>
    <Header>
      <SearchForm />
    </Header>
    <div className="container">
      <FilmsList />
    </div>
    <Footer />
  </ErrorBoundary>
);
