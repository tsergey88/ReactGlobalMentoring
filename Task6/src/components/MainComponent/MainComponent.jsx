import React from "react";

import { Header } from "../Header";
import { Footer } from "../Footer";
import { FilmsList } from "../FilmsList";
import { ErrorBoundary } from "../ErrorBoundary";
import { SearchForm } from "../SearchForm";

export const MainComponent = ({ ...props }) => (
  <ErrorBoundary>
    <Header>
      <SearchForm {...props} />
    </Header>
    <div className="container">
      <FilmsList {...props} />
    </div>
    <Footer />
  </ErrorBoundary>
);
