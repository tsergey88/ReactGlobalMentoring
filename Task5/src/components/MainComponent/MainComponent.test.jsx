import React from "react";
import { MainComponent } from "./MainComponent";
import { shallow } from "enzyme";

describe("MainComponent component", () => {
  const component = shallow(<MainComponent />);
  it("should be render", () => {
    expect(component).toMatchSnapshot();
  });
});
