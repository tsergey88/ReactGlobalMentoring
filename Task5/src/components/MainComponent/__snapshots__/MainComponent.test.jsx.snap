// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent component should be render 1`] = `
<ErrorBoundary>
  <Header>
    <SearchForm />
  </Header>
  <div>
    <StateLine />
    <FilmsList />
  </div>
  <Footer />
</ErrorBoundary>
`;
